internal/distributed/proxy/httpserver/handler_v2.go

```diff
@@ -905,61 +905,50 @@ func generatePlaceholderGroup(ctx context.Context, body string, collSchema *sche
 	})
 }
 
-func generateSearchParams(ctx context.Context, c *gin.Context, reqParams map[string]float64) ([]*commonpb.KeyValuePair, error) {
-	params := map[string]interface{}{ // auto generated mapping
-		"level": int(commonpb.ConsistencyLevel_Bounded),
-	}
-	if reqParams != nil {
-		radius, radiusOk := reqParams[ParamRadius]
-		rangeFilter, rangeFilterOk := reqParams[ParamRangeFilter]
-		if rangeFilterOk {
-			if !radiusOk {
-				log.Ctx(ctx).Warn("high level restful api, search params invalid, because only " + ParamRangeFilter)
-				HTTPAbortReturn(c, http.StatusOK, gin.H{
-					HTTPReturnCode:    merr.Code(merr.ErrIncorrectParameterFormat),
-					HTTPReturnMessage: merr.ErrIncorrectParameterFormat.Error() + ", error: invalid search params",
-				})
-				return nil, merr.ErrIncorrectParameterFormat
-			}
-			params[ParamRangeFilter] = rangeFilter
-		}
-		if radiusOk {
-			params[ParamRadius] = radius
-		}
-	}
-	bs, _ := json.Marshal(params)
-	searchParams := []*commonpb.KeyValuePair{
-		{Key: Params, Value: string(bs)},
-	}
+func generateSearchParams(ctx context.Context, c *gin.Context, reqSearchParams searchParams) ([]*commonpb.KeyValuePair, error) {
+	var searchParams []*commonpb.KeyValuePair
+	bs, _ := json.Marshal(reqSearchParams.Params)
+	searchParams = append(searchParams, &commonpb.KeyValuePair{Key: Params, Value: string(bs)})
+	searchParams = append(searchParams, &commonpb.KeyValuePair{Key: common.IgnoreGrowing, Value: strconv.FormatBool(reqSearchParams.IgnoreGrowing)})
+	// need to exposure ParamRoundDecimal in req?
+	searchParams = append(searchParams, &commonpb.KeyValuePair{Key: ParamRoundDecimal, Value: "-1"})
 	return searchParams, nil
 }
 
 func (h *HandlersV2) search(ctx context.Context, c *gin.Context, anyReq any, dbName string) (interface{}, error) {
 	httpReq := anyReq.(*SearchReqV2)
 	req := &milvuspb.SearchRequest{
-		DbName:                dbName,
-		CollectionName:        httpReq.CollectionName,
-		Dsl:                   httpReq.Filter,
-		DslType:               commonpb.DslType_BoolExprV1,
-		OutputFields:          httpReq.OutputFields,
-		PartitionNames:        httpReq.PartitionNames,
-		UseDefaultConsistency: true,
+		DbName:         dbName,
+		CollectionName: httpReq.CollectionName,
+		Dsl:            httpReq.Filter,
+		DslType:        commonpb.DslType_BoolExprV1,
+		OutputFields:   httpReq.OutputFields,
+		PartitionNames: httpReq.PartitionNames,
+	}
+	if httpReq.ConsistencyLevel != "" {
```

Comment:
Is it possible to extract this and reuse at below line [997](https://github.com/milvus-io/milvus/pull/36304/files#diff-b81c4eb20e49c64dc253ff54abb0a4d56aace2d6e63fd78b9be98f312c9a678bR997)